util/lua/lua_test.go

```diff
@@ -369,8 +386,27 @@ func TestExecuteResourceActionDiscovery(t *testing.T) {
 				Name: "replicas",
 				Type: "number",
 			}},
+		},
+	}
+	for _, expectedAction := range expectedActions {
+		assert.Contains(t, actions, expectedAction)
+	}
+}
+
+func TestExecuteResourceActionDiscoveryWithDuplicationActions(t *testing.T) {
+	testObj := StrToUnstructured(objJSON)
+	vm := VM{}
+	actions, err := vm.ExecuteResourceActionDiscovery(testObj, []string{validDiscoveryLua, validDiscoveryLua})
```

Comment:
You are using the same discovery script for your test. In the code, the array order has a major significance. 
The test should validate that the ExecuteResourceActionDiscovery returns the correct results.
- Based on array order
- For duplicated action
- For action in script 0, not present in script 1
- For action in script 1, not present in script 0

Reply:
array order should not matter, as code loops through the script and only store the unique actions. Reason i'm passing same lua script is to test the unique action, incase the action are coming from upstream and same action is provided in overrides.  Code would just pick the first action and drop the  second duplicate action. This change only deal with merge and doesn't provide any option to override specific action.  UI further sort in asc irrespective of the given order.